### PR TITLE
Converts the RnD console to NanoUI

### PR DESCRIPTION
**Pull request by printer16**
This makes the RnD console use NanoUI. 
DNM for now.
Putting it up so if I'm forgetting something/doing something wrong I fix it now rather than have to go back and fix more of it once I finish.

### CODE REVIEW

File: code/modules/research/rdconsole.dm [hunk 1]
```diff
@@ -145,6 +145,111 @@ won't update every console in existence) but it's more of a hassle to do. Also,
 		user << "<span class='notice'>You you disable the security protocols.</span>"
 		return 1
 
+/obj/machinery/computer/rdconsole/proc/get_task()
+	var/task = null
+	switch(screen)
+		if(0.0)
```
Comment (skull132) [inline code/modules/research/rdconsole.dm]: Don't use decimals. Use 10s instead.

So `0.0` becomes `10`, `0.1` -> `11` etc, `1.0` -> `20` etc. The reason is that integers are preferred over floating point numbers whenever possible, as they're less odd to handle.

File: code/modules/research/rdconsole.dm [hunk 2]
```diff
@@ -145,6 +145,111 @@ won't update every console in existence) but it's more of a hassle to do. Also,
...
+	var/data[0]
+	var/task = get_task()
+	var/textscreen = screen_number2text()
+	var/knowndesigns[0]
+	var/linked_lathe_queue[0]
+	for(var/datum/design/D in linked_lathe.queue)
+		linked_lathe_queue.Add(list(list("QName" = "[D.name]")))
+
+	for(var/datum/design/D in files.known_designs)
+		knowndesigns.Add(list(list("Name" = "[D.name]", "BuildType" = "[D.build_type]", "BuildPath" = "[D.build_path]", "BuildTime" = "[D.time]", "ID" = "[D.id]")))
```
Comment (skull132) [inline code/modules/research/rdconsole.dm]: `knowndesigns += list(list())`

File: code/modules/research/rdconsole.dm [hunk 3]
```diff
@@ -145,6 +145,111 @@ won't update every console in existence) but it's more of a hassle to do. Also,
...
+	return textscreen
+
+/obj/machinery/computer/rdconsole/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	var/data[0]
+	var/task = get_task()
+	var/textscreen = screen_number2text()
+	var/knowndesigns[0]
+	var/linked_lathe_queue[0]
+	for(var/datum/design/D in linked_lathe.queue)
+		linked_lathe_queue.Add(list(list("QName" = "[D.name]")))
```
Comment (skull132) [inline code/modules/research/rdconsole.dm]: Same as below.

File: nano/templates/rdconsole.tmpl [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        {{:helper.link('Main Menu', 'gear', {'menu' : 1},(data.screen == 1 ? 'selected' : null), 'fixedLeftWider')}}
+        {{:helper.link('Destructive Analyzer Control','gear', {'menu' :((data.linked_analyzer == null) ? 2 : 2.1)}, ((data.screen >= 2 && data.screen < 2.9) ? 'selected' : null),'fixedLeftWider')}}
+        {{:helper.link('Protolathe Control', 'gear', {'menu' :((data.linked_lathe == null) ? 3 : 3.1)}, ((data.screen >= 3 && data.screen < 3.9) ? 'selected' : null),'fixedLeftWider')}}
+        {{:helper.link('Circuit Imprinter Control', 'gear', {'menu' :((data.linked_imprinter == null) ? 4 : 4.1)}, ((data.screen >= 4 && data.screen < 4.9) ? 'selected' : null),'fixedLeftWider')}}
+        {{:helper.link('Device Management', 'gear', {'menu' : 5}, ((data.screen >= 5 && data.screen < 5.9) ? 'selected' : null),'fixedLeftWider')}}
+	</div>
+</div>
+{{if data.screen}}
+    <center><H3>Current Menu</span>: {{:data.textscreen}}</H3>
+    {{if data.screen == 1.0}}
```
Comment (skull132) [inline nano/templates/rdconsole.tmpl]: All of these `if data.screen`s should be in one `if/elif/elif/elif/else` chain.

**Comment by printer16** [comment]
Will be redone in the future.

**Pull request closed without merging**